util/filesort/filesort.go

```diff
@@ -560,11 +557,6 @@ func (w *Worker) input(row *comparableRow) error {
 		w.ctx.external = true
 		go w.flushToFile()
 	}
-
-	if w.err != nil {
```

Comment:
the error is set on the background, we don't need to check it here.
It will be checked later when Output is called.